vtex/toolbelt
# Download own app types instead of adding it to package.json

- Download own app types instead of adding it to `package.json`
- Change `vteximg` URL to `vtexassets`, as required from io-infra-team
- Add new options to `vtex setup`: 
  - `--types`: setup only types
  - `--own-types`: setup only the types for the app itself
- Refactor types setup to use more of an OOP approach (open to suggestions to improve the design)
- Add a `postinstall` script on react and node builders `package.json`, if they are present: The `postinstall` script will run when `yarn add`, `yarn install` or `yarn` are executed. Unfortunately I didn't find a way to run it when `yarn remove`. The script is set to `vtex setup --types --own-types`, so after the yarn command completes the types bundle for the app is downloaded and added to the `node_modules`. This is necessary because `yarn` uses the `package.json` as source of truth to the `node_modules`, and when it installs or removes something, the `node_modules` is reset.

  
#### What problem is this solving?
The workflow for developing a Node + GraphQL app is somewhat complicated. When developing, the URL that was inserted on `package.json` regarding the app GraphQL types were a linked app URL. Before publishing and merging the GitHub branch, the developer had to remove this URL to prevent a linked app URL to go to master, but after publishing they had to run `vtex setup --ignore-linked` to add the published URL types to `package.json`. This way the app types on `package.json` would maintain consistency.

This problem would be solved if the types for itself weren't added on `package.json`. 

#### How should this be manually tested?

Install this branch:
```
git clone https://github.com/vtex/toolbelt.git && \
cd toolbelt && \
git checkout feat/refactor-types && \
yarn && yarn global add file:$PWD
```
Run `vtex setup` in some apps, linked included, and check if the types for the app itself are on `node_modules`, but not in `package.json`.

#### Types of changes
